### PR TITLE
[IMP] website, website_*: improve edit in backend feature

### PR DESCRIPTION
**Pull request by rdeodoo**
With this commit:
1. It is now possible to edit in backend any desired model, not only the ones
   which have the published mixin.
   This commit add the Edit in backend for Forum and Blog.
2. It is now possible to land on another module in backend than Website.
   That's especially useful for events and slides which are not directly
   related to the website module as they have their own module.
3. Remove the custom Edit in backend from the forum homepage (fa-cog).
   Opportunity was also taken to remove the change the edit welcome message.
   It will now be editable through the editor (welcome message will appear in
   edit mode). Thus we got rid of the custom edit welcome message controller
   and views.

### CODE REVIEW

File: addons/website/models/ir_ui_view.py [hunk 1]
```diff
@@ -332,6 +332,7 @@ def _prepare_qcontext(self):
             translatable = editable and self._context.get('lang') != request.env['ir.http']._get_default_lang().code
             editable = not translatable and editable
 
+            # TODO (review) this is called before the dict update with the controller values so it will never contain a main object?
```
Comment (JKE-be) [inline addons/website/models/ir_ui_view.py]: ok go for it 
```

 qcontext.update(dict(
                main_object=self,
```

File: odoo/addons/base/models/ir_ui_view.py [hunk 2]
```diff
@@ -1149,6 +1149,10 @@ def render(self, values=None, engine='ir.qweb', minimal_qcontext=False):
         qcontext = dict() if minimal_qcontext else self._prepare_qcontext()
         qcontext.update(values or {})
 
+        if 'main_object' in qcontext:
+            func = getattr(qcontext['main_object'], 'get_backend_menu_id', False)
+            qcontext['backend_menu_id'] = func and func() or self.env.ref('website.menu_website_configuration').id
```
Comment (JKE-be) [inline odoo/addons/base/models/ir_ui_view.py]: move me in httprout/webs

File: addons/website_blog/views/website_blog_templates.xml [hunk 3]
```diff
@@ -30,6 +30,7 @@
 ============================================================================ -->
 <template id="index" name="Blog Navigation">
     <t t-call="website.layout">
+        <t t-if="main_object._name in ('blog.blog', 'blog.post')" t-set="edit_in_backend" t-value="True"/>
```
Comment (JKE-be) [inline addons/website_blog/views/website_blog_templates.xml]: move me in main controller of blog.

File: addons/website/models/ir_ui_view.py (hunk 1 above)
Comment (rdeodoo) [reply addons/website/models/ir_ui_view.py]: :+1: done

File: odoo/addons/base/models/ir_ui_view.py (hunk 2 above)
Comment (rdeodoo) [reply odoo/addons/base/models/ir_ui_view.py]: Done

File: addons/website_forum/controllers/main.py [hunk 4]
```diff
@@ -123,6 +123,7 @@ def questions(self, forum, tag=None, page=1, filters='all', sorting=None, search
         values = self._prepare_user_values(forum=forum, searches=post, header={'ask_hide': not forum.active})
         values.update({
             'main_object': tag or forum,
```
Comment (rdeodoo) [inline addons/website_forum/controllers/main.py]: Moving edit_in_backend in controller will now show edit in backend for tag.
I don't see any problem with that but if you don't want it tell me, I'll only add it for forum.

Comment (rdeodoo) [reply addons/website_forum/controllers/main.py]: Only for forum, done

**Comment by rdeodoo** [comment]
Rebased

**Comment by rdeodoo** [comment]
@JKE-be It now has conflicts, is that something we want for oxp or we will merge it next week? Still 12.5 or master?

**Comment by rdeodoo** [comment]
Conflicts fixes & rebased.
Don't forget to r+ https://github.com/odoo/enterprise/pull/5725

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by qsm-odoo** [comment]
@robodoo rebase-ff

**Comment by JKE-be** [comment]
@robodoo retry r+

**Comment by JKE-be** [comment]
bha alors merge la au lieu de te plaindre @robodoo !

**Comment by rdeodoo** [comment]
Had conflicts with the welcome message that this PR is removing since the PR about forum revamp changed it
Should be good now, tested it.

**Comment by JKE-be** [comment]
@robodoo r+

**Pull request closed without merging**